documentation/modules/ROOT/pages/connectors/db2.adoc

```diff
@@ -217,12 +428,51 @@ If the default topic name do not meet your requirements, you can configure custo
 
+// Type: concept
+// ModuleID: how-debezium-db2-connectors-handle-database-schema-changes
+// Title: How {prodname} Db2 connectors handle database schema changes
+[[db2-schema-history-topic]]
+=== Schema history topic
+
+When a database client queries a database, the client uses the database’s current schema.
+However, the database schema can be changed at any time, which means that the connector must be able to identify what the schema was at the time each insert, update, or delete operation was recorded.
+Also, a connector cannot necessarily apply the current schema to every event.
+If an event is relatively old, it's possible that it was recorded before the current schema was applied.
+
+////
+{prodname}requires the schema history so that it can apply the table structure in effect at the time an event occurs.
+When the connector is stopped and restarted if the database metadata has changed so that the current structure of the table does not match the structure that was in effect at the point in time we're reading changes from; which is why the topic is needed.
+
+The connector reads offsets at start-up if they exist.
+It then uses the position to recover the schema history topic, but only up to the position specified by the offsets.
+
+If you apply schema changes to a table after the position of the last event that is recorded in the offsets, when the connector restarts, it starts from its last known position, which, according to the offsets file, excludes the most recent schema data in the dbhistory file
+It ignores the subsequent schema changes, skipping over them without reporting their presence.
+So the connector tries to process the event at the next offset without having the schema data.
+
+////
+
+To ensure correct processing of changes that occur after a schema change, Db2 includes in the transaction log not only the row-level changes that affect the data, but also the DDL statements that are applied to the database.
+As the connector reads the transaction log and comes across these DDL statements, it parses them and updates an in-memory representation of each table’s schema.
+The connector uses this schema representation to identify the structure of the tables at the time of each insert, update, or delete operation and to produce the appropriate change event.
+In a separate database schema history Kafka topic, the connector records all DDL statements along with the position in the transaction log where each DDL statement appeared.
+
+When the connector restarts after either a crash or a graceful stop, it starts reading the transaction log from a specific position, that is, from a specific point in time.
+The connector rebuilds the table structures that existed at this point in time by reading the database schema history Kafka topic and parsing all DDL statements up to the point in the transaction log where the connector is starting.
+
```

Comment:
```suggestion
The connector rebuilds the table structures that existed at this point in time by reading the database schema history Kafka topic up to the point in the transaction log where the connector is starting.

```
We likely want to avoid the use of parsing here because that's not necessarily accurate.  For Db2 specifically, we use the previously cached schema representation that does not include any DDL at all.  For SQL Server we use the same strategy.  I'd probably just be consistent with removing this across all conenctors.